ci-automation/garbage_collect.sh

```diff
@@ -132,4 +132,12 @@ function garbage_collect() {
             echo "## (DRY_RUN=y so not doing anything) ##"
         fi
     done
+
+    ## Let's clear the cloud provider resources created during the tests.
+    if [ "$dry_run" != "y" ] ; then
+        vendors=($(find ci-automation/vendor-testing -not -name "qemu*" -type f -printf "%f "))
+        for vendor in "${vendors[@]}"; do
+            ore "${vendor%.*}" gc --duration 2h
```

Comment:
Now I wonder how that works at the moment; is 2 hours not a bit too short for Equinix Metal tests, now that we have the reuse logic for instances?

Looks good, but the for loop also needs to pass the creds, as far as I know.

Reply:
Don't you think we could add this logic right _after_ the tests in the test job ?

Once all tests have ran for a provider, we cleanup resources for this very provider with ore - in this context, we still have access to the credentials.

Reply:
The gc job should be run periodically. Just running it after the job would mean that over the weekend where no changes are pushed it would not run. It would also mean that it won't run for platforms where we trigger the tests manually and not nightly.

Reply:
I get your points. For the record, I just mentioned moving the garbage collection of the cloud provider resources not the other stuffs that we would keep as they are.

Tests should most of the time be CI only and a test script would:
* create the resources
* consume the resources
* delete the created resources

For manual testing, we should use the exact same scripts so if we garbage collect from the test script directly we should be good.

For manual testing _without_ using the scripts, it should up to the user to delete the created resources IMHO.

Reply:
The garbage collection is different from the resource cleanup that kola does itself (or that the job does with `ore` to remove the uploaded image).
The cloud resource GC job should run periodically, for example, in the case where Jenkins job got terminated due to the Jenkins nodes rebooting while jobs were running. The leftover cloud resources would accumulate costs or in distrub execution of other tests because the instance limits are hit.

Reply:
I filed a PR here now: https://github.com/flatcar-linux/scripts/pull/306

You can drop this code because it would anyway not find `ore` (this code directly runs on Flatcar)